src/views/Dashboard/components/Widgets/BlocksWidget/BlocksWidget.tsx

```diff
@@ -1,133 +1,177 @@
-    numeric: true,
-    disablePadding: false,
     label: "Tx Qty",
+    headAlignment: "center",
+    rowAlignment: "center",
   },
   {
     id: "value",
-    numeric: true,
-    disablePadding: false,
     label: "Value",
+    headAlignment: "center",
+    rowAlignment: "center",
   },
   {
     id: "generator",
-    numeric: true,
-    disablePadding: false,
     label: "Generator",
+    headAlignment: "center",
+    rowAlignment: "center",
   },
   {
     id: "baseTarget",
-    numeric: true,
-    disablePadding: false,
     label: "Base Target",
+    headAlignment: "center",
+    rowAlignment: "center",
   },
 ];
 
+interface IBlockDetail {
+  height: number;
+  headers: Array<IHeadCellProps>;
+  rows: Array<ITableRow>;
+}
+
 const BlocksWidget: React.FC = () => {
+  const [detailedDialogOpen, setDetailedDialogOpen] = useState(false);
+  const [blockDetail, setBlockDetail] = useState<IBlockDetail | undefined>(undefined);
   const { recentBlocks } = useBlocks();
 
-  let blockRows;
+  const handleOpenBlockDetail = useCallback(
+    (height: number) => {
+      const block = recentBlocks?.filter((x) => x.height === height)[0];
 
-  if (recentBlocks) {
-    blockRows = recentBlocks.map((row, index) => {
-      return (
-        // Slide transition isn't working quite yet. It does work on initial page load and when toggling row count
-        //  so maybe a memo needs to be involved to get this working for new rows. Maybe comparing a useRef?
-        <TransitionGroup key={"tg-" + index} component={null}>
-          <Slide direction="left" timeout={DefaultTransitionTime} appear={true}>
-            <TableRow hover tabIndex={-1} key={row.timestamp + "-" + index}>
-              <TableCell align="right">{row.height}</TableCell>
-              <TableCell align="right">
-                {new Date(row.timestamp * 1000 + JUPGenesisTimestamp * 1000).toLocaleString(userLocale.localeStr, userLocale.options)}
-              </TableCell>
-              <TableCell align="right">{row.numberOfTransactions}</TableCell>
-              <TableCell align="right">{row.totalAmountNQT}</TableCell>
-              <TableCell align="right">{row.generatorRS}</TableCell>
-              {/* baseTarget row is not ideal but it fixes an upstream calculation error due to JUP's block time changes over time */}
-              <TableCell align="right">{Math.round(parseInt(row.baseTarget) / 153722867 / 10) + " %"}</TableCell>
-            </TableRow>
-          </Slide>
-        </TransitionGroup>
-      );
-    });
-  }
+      setDetailedDialogOpen(true);
+      setBlockDetail({
+        height,
+        headers: [
+          {
+            id: "col1",
+            label: "Name",
+            headAlignment: "center",
+            rowAlignment: "center",
+          },
+          {
+            id: "col2",
+            label: "Details",
+            headAlignment: "center",
+            rowAlignment: "center",
+          },
+        ],
+        rows: [
+          {
```

Comment:
now that we know what we want displayed here, it should be trivial in the future to cleanup this with a mapping or a loop.